tests/cover/test_cathetus.py

```diff
@@ -0,0 +1,193 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2018 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import sys
+import math
+
+from hypothesis.internal.cathetus import cathetus
+
+
+def assert_cathetus_exact(h, a, b):
+    b0 = cathetus(h, a)
+    assert b == b0, (
+        'expected cathetus(%g, %g) == %g, got %g' %
+        (h, a, b, b0)
+    )
+
+
+def assert_cathetus_nan(h, a):
+    b = cathetus(h, a)
+    assert math.isnan(b), (
+        'expected cathetus(%g, %g) == NaN, got %g' %
+        (h, a, b)
+    )
+
+
+def assert_cathetus_inf(h, a):
+    b = cathetus(h, a)
+    assert math.isinf(b), (
+        'expected cathetus(%g, %g) == Infinity, got %g' %
+        (h, a, b)
+    )
+
+
+def test_cathetus_subnormal_underflow():
+    u = sys.float_info.min * sys.float_info.epsilon
+    h = 5 * u
+    a = 4 * u
+    b = cathetus(h, a)
+    assert not math.isnan(b), (
+        'expecting cathetus(%g, %g) not NaN, got %g' %
+        (h, a, b)
+    )
+    assert b > 0, (
+        'expecting cathetus(%g, %g) positive, got %g' %
+        (h, a, b)
+    )
+    assert b == 3 * u, (
+        'expecting cathetus(%g, %g) == %g, got %g' %
+        (h, a, 3 * u, b)
+    )
+
+
+def test_cathetus_simple_underflow():
+    a = sys.float_info.min
+    h = a * math.sqrt(2)
+    b = cathetus(h, a)
+    assert not math.isnan(b), (
+        'expecting cathetus(%g, %g) not NaN, got %g' %
+        (h, a, b)
+    )
+    assert b > 0, (
+        'expecting cathetus(%g, %g) positive, got %g' %
+        (h, a, b)
+    )
+
+
+def test_cathetus_huge_no_overflow():
+    h = sys.float_info.max
+    a = h / math.sqrt(2)
+    b = cathetus(h, a)
+    assert not (math.isinf(b) or math.isnan(b)), (
+        'expecting cathetus(%g, %g) finite, got %g' %
+        (h, a, b)
+    )
+
+
+def test_cathetus_large_no_overflow():
+    h = sys.float_info.max / 3
+    a = h / math.sqrt(2)
+    b = cathetus(h, a)
+    assert not (math.isinf(b) or math.isnan(b)), (
+        'expecting cathetus(%g, %g) finite, got %g' %
+        (h, a, b)
+    )
+
+
+def test_cathetus_nan_hypot():
+    assert_cathetus_nan(float(u'nan'), 3)
+    assert_cathetus_nan(float(u'nan'), -3)
+    assert_cathetus_nan(float(u'nan'), 0)
+    assert_cathetus_nan(float(u'nan'), float(u'inf'))
+    assert_cathetus_nan(float(u'nan'), float(u'nan'))
```

Comment:
Repetitive tests like this should be defined with `@pytest.mark.parametrize`, to isolate individual test cases.